File: testdata/fixtures/bench/javascript/vulnerable/a10_ssrf.ts
// Source: OWASP Juice Shop - SSRF via profile image URL
// Expected: GTSS-SSRF-001 (URL from User Input)
// OWASP: A10:2021 - Server-Side Request Forgery

import { Request, Response } from 'express';
import http from 'http';
import https from 'https';

export function fetchProfileImage(req: Request, res: Response): void {
  const imageUrl = req.body.imageUrl;
  const client = imageUrl.startsWith('https') ? https : http;
  client.get(imageUrl, (proxyRes) => {
    res.setHeader('Content-Type', proxyRes.headers['content-type'] || 'image/png');
    proxyRes.pipe(res);
  }).on('error', (err) => {
    res.status(500).json({ error: 'Failed to fetch image' });
  });
}

export function webhookProxy(req: Request, res: Response): void {
  const targetUrl = req.query.url as string;
  const options = new URL(targetUrl);
  http.get(options, (proxyRes) => {
    let data = '';
    proxyRes.on('data', (chunk) => data += chunk);
    proxyRes.on('end', () => res.json({ response: data }));
  });
}

File: testdata/fixtures/bench/go/vulnerable/a10_ssrf.go
// Source: CWE-918 - Server-Side Request Forgery in Go
// Expected: GTSS-SSRF-001 (URL from User Input)
// OWASP: A10:2021 - Server-Side Request Forgery

package handler

import (
	"io"
	"net/http"
)

func ProxyFetch(w http.ResponseWriter, r *http.Request) {
	targetURL := r.URL.Query().Get("url")
	resp, err := http.Get(targetURL)
	if err != nil {
		http.Error(w, "Fetch failed", http.StatusBadGateway)
		return
	}
	defer resp.Body.Close()
	w.Header().Set("Content-Type", resp.Header.Get("Content-Type"))
	io.Copy(w, resp.Body)
}

func WebhookForward(w http.ResponseWriter, r *http.Request) {
	webhookURL := r.FormValue("callback_url")
	payload := r.FormValue("data")
	client := &http.Client{}
	req, _ := http.NewRequest("POST", webhookURL, nil)
	req.Header.Set("Content-Type", "application/json")
	_ = payload
	resp, err := client.Do(req)
	if err != nil {
		http.Error(w, "Forward failed", http.StatusBadGateway)
		return
	}
	defer resp.Body.Close()
	io.Copy(w, resp.Body)
}
